src/app/newSettings/Sections/Workspace/Teams/TeamsSection.tsx

```diff
@@ -215,10 +236,84 @@ const TeamsSection = ({ onClosePreferences }: { onClosePreferences: () => void }
     }
   };
 
+  const deleteTeam = async () => {
+    setIsDeleteTeamLoading(true);
+    try {
+      if (selectedWorkspace && selectedTeam) {
+        await workspacesService.deleteTeam(selectedWorkspace.workspaceUser.workspaceId, selectedTeam.team.id);
+      }
+      setSelectedTeam(null);
+      setIsDeleteTeamDialogOpen(false);
+      setTimeout(() => {
+        getTeams();
+      }, 500);
+    } catch (err) {
+      const castedError = errorService.castError(err);
+      errorService.reportError(castedError);
+    }
+    setIsDeleteTeamLoading(false);
+  };
+
+  const removeTeamMember = async () => {
+    setIsRemoveTeamMemberLoading(true);
+    try {
+      if (selectedTeam && teamMemberToRemove) {
+        await workspacesService.removeTeamUser(selectedTeam.team.id, teamMemberToRemove?.uuid);
+      }
+      setTimeout(() => {
+        getWorkspacesMembers();
+        selectedTeam && getTeamMembers(selectedTeam);
+        getTeams();
+      }, 500);
+    } catch (err) {
+      const castedError = errorService.castError(err);
+      errorService.reportError(castedError);
+    }
+    setIsRemoveTeamMemberLoading(false);
+    setIsRemoveTeamMemberDialogOpen(false);
+    setTeamMemberToRemove(null);
+  };
+
+  const changeManager = async () => {
+    setIsChangeManagerLoading(true);
+    try {
+      if (selectedWorkspace && selectedTeam && newTeamManager) {
+        await workspacesService.changeTeamManager(
+          selectedWorkspace.workspaceUser.workspaceId,
+          selectedTeam.team.id,
+          newTeamManager.uuid,
+        );
+      }
+      setTimeout(() => {
+        getTeams();
+        getWorkspacesMembers();
+        selectedTeam && getTeamMembers(selectedTeam);
+      }, 500);
+    } catch (err) {
+      const castedError = errorService.castError(err);
+      errorService.reportError(castedError);
+    }
+    setIsChangeManagerLoading(false);
+    setIsChangeManagerDialogOpen(false);
+  };
+
+  const handleChangeManagerClicked = (member: TeamMember) => {
+    setNewTeamManager(member);
+    selectedTeamMembers.map((member) => member.uuid === selectedTeam?.team.managerId && setCurrentTeamManager(member));
+    setIsChangeManagerDialogOpen(true);
```

Comment:
Maybe you can create containers for each modal and add the specific logic for each one there, so everything is more modular and can be better maintained

Reply:
Done!
The only one I haven't created a container for is the teams list because it didn't have its own logic.